### DISCUSSION TITLE
Help getting express.js bodyparser to work once deployed to vercel

### DISCUSSION THREAD

**Discussion by mriely** [vercel/community#7016] [Help]
### Summary

Hello, I'm trying to make an express.js app. I'm using the template and I adapted it for my use and it worked for a while. I keep going all the way back to the template to try to get a deployment to be able to parse the body of a POST. 

From the example project, here is how I'm trying to parse:

`app.post('/uploadSuccessful', urlencodedParser, async (req, res) => {`

Here is the beginning of the index.ts file. This is unmodified from the template. I added the example users table to my database and it works locally. On production I get nulls into the database because urlencodedParser is not working.

`require('dotenv').config();

const express = require('express');
const app = express();
const { sql } = require('@vercel/postgres');

const bodyParser = require('body-parser');
const path = require('path');

// Create application/x-www-form-urlencoded parser
const urlencodedParser = bodyParser.urlencoded({ extended: false });`

### Steps to Reproduce

Use the example  from examples/solutions/express, edit the database credentials, run vercel --prod , use postman to post

**Comment by QuiiBz** [comment]
This template uses the `urlencoded` parser on the `uploadSuccessful` endpoint, which means you need to set a `Content-Type: application/x-www-form-urlencoded` header to your POST request, and that your body needs to follow the correct format: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST#example

If you want to POST JSON data, you can switch to `bodyParser.json()` instead of `bodyParser.urlencoded({ extended: false })`. Or if you want plain text, you can switch to `bodyParser.text()`.

You can also try locally your application (`vercel dev` or `npm run start`) before deploying to make sure you fixed the issue.
